community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -971,22 +1110,27 @@ public PrimitiveIntIterator graphGetPropertyKeys( KernelStatement state )
     }
 
     @Override
-    public Property graphGetProperty( KernelStatement state, int propertyKeyId )
+    public boolean graphHasProperty( KernelStatement state, int propertyKeyId )
+    {
+        return graphGetProperty( state, propertyKeyId ) != null;
+    }
```

Comment:
Or does the API look like this to open up for more efficient implementations in the future without having to change API again?